src/gui/transaction_test.go

```diff
@@ -0,0 +1,794 @@
+package gui
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"math"
+
+	"time"
+
+	"net/url"
+
+	"errors"
+
+	"bytes"
+	"encoding/hex"
+
+	"github.com/stretchr/testify/mock"
+
+	"github.com/skycoin/skycoin/src/cipher"
+	"github.com/skycoin/skycoin/src/coin"
+	"github.com/skycoin/skycoin/src/daemon"
+	"github.com/skycoin/skycoin/src/testutil"
+	"github.com/skycoin/skycoin/src/util/utc"
+	"github.com/skycoin/skycoin/src/visor"
+)
+
+func createUnconfirmedTxn(t *testing.T) visor.UnconfirmedTxn {
+	ut := visor.UnconfirmedTxn{}
+	ut.Txn = coin.Transaction{}
+	ut.Txn.InnerHash = testutil.RandSHA256(t)
+	ut.Received = utc.Now().UnixNano()
+	ut.Checked = ut.Received
+	ut.Announced = time.Time{}.UnixNano()
+	return ut
+}
+
+func makeTransaction(t *testing.T) coin.Transaction {
+	tx, _ := makeTransactionWithSecret(t)
+	return tx
+}
+
+func makeUxOutWithSecret(t *testing.T) (coin.UxOut, cipher.SecKey) {
+	body, sec := makeUxBodyWithSecret(t)
+	return coin.UxOut{
+		Head: coin.UxHead{
+			Time:  100,
+			BkSeq: 2,
+		},
+		Body: body,
+	}, sec
+}
+
+func makeAddress() cipher.Address {
+	p, _ := cipher.GenerateKeyPair()
+	return cipher.AddressFromPubKey(p)
+}
+
+func makeTransactionWithSecret(t *testing.T) (coin.Transaction, cipher.SecKey) {
+	tx := coin.Transaction{}
+	ux, s := makeUxOutWithSecret(t)
+
+	tx.PushInput(ux.Hash())
+	tx.SignInputs([]cipher.SecKey{s})
+	tx.PushOutput(makeAddress(), 1e6, 50)
+	tx.PushOutput(makeAddress(), 5e6, 50)
+	tx.UpdateHeader()
+	return tx, s
+}
+
+func makeUxBodyWithSecret(t *testing.T) (coin.UxBody, cipher.SecKey) {
+	p, s := cipher.GenerateKeyPair()
+	return coin.UxBody{
+		SrcTransaction: testutil.RandSHA256(t),
+		Address:        cipher.AddressFromPubKey(p),
+		Coins:          1e6,
+		Hours:          100,
+	}, s
+}
+
+func TestGetPendingTxs(t *testing.T) {
+	invalidTxn := createUnconfirmedTxn(t)
+	invalidTxn.Txn.Out = append(invalidTxn.Txn.Out, coin.TransactionOutput{
+		Coins: math.MaxInt64 + 1,
+	})
+
+	tt := []struct {
+		name                          string
+		method                        string
+		url                           string
+		status                        int
+		error                         string
```

Comment:
use `err` not `error`, otherwise syntax highlighting is wrong because `error` has special meaning